### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by sagieramos**
### Project requirements
- Remove `current_user` method in `ApplicationController`, devise will provide us with one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb]: - [ ] @sagieramos and @Samuelwanza Great work adding form to register new users, however the styling of this page doesn't match the wireframe design given, check the comparison below:

Wireframe Design          |  Your design     
:-------------------------:|:-------------------------:
image  | image

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/sessions/new.html.erb]: - [ ] @sagieramos and @Samuelwanza Great work adding a form to login  users, however, the styling of this page doesn't match the wireframe design given, check the comparison below:

Wireframe Design          |  Your design     
:-------------------------:|:-------------------------:
image  | image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @sagieramos and @Samuelwanza , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added Devise correctly and removed the current method.
✔️ Ensured the users are able to register and login into the app.

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (sagieramos) [reply app/views/devise/registrations/new.html.erb]: I appreciate the time you took to review my code thoroughly. Even though the wireframe of the sign_up page isn't given, the field in the User model to add a new user requires a minimum of 3 values (name, email, and password). This makes it impossible to make the sign_up wireframe look the same as the sign_in. 

Screenshot from 2023-12-14 12-44-33

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (sagieramos) [reply app/views/devise/sessions/new.html.erb]: Thanks for pointing this out. I have made the necessary changes by replacing the placeholder of each field with its supposed label. 
In other to make the UI look the same as the wireframe, I have removed the signup button used in adding a new user. The sign_up button will be shown in the UI when a sign_in fails due to credentials not being present in the database

**Review by gemmen29 — APPROVED** [review]
Hi @sagieramos, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**